static/app/views/alerts/list/rules/teamFilter.tsx

```diff
@@ -1,119 +1,119 @@
-  const isSuperuser = isActiveSuperuser();
+  const {teams, onSearch, fetching} = useTeams({provideUserTeams: showIsMemberTeams});
 
-  const additionalOptions = [
-    {
-      label: t('My Teams'),
-      value: 'myteams',
-      checked: selectedTeams.has('myteams'),
-      filtered: false,
-    },
-    {
-      label: t('Unassigned'),
-      value: 'unassigned',
-      checked: selectedTeams.has('unassigned'),
-      filtered: false,
-    },
-  ];
-  const isMemberTeams = teams.filter(team => team.isMember);
-  const teamItems = (isSuperuser ? teams : showIsMemberTeams ? isMemberTeams : teams).map(
-    ({id, slug}) => ({
-      label: slug,
-      value: id,
-      filtered: teamFilterSearch
-        ? !slug.toLowerCase().includes(teamFilterSearch.toLowerCase())
-        : false,
-      checked: selectedTeams.has(id),
-    })
+  const teamOptions = useMemo(
+    () =>
+      teams.map(team => ({
+        value: team.id,
+        label: `#${team.slug}`,
+        leadingItems: <TeamAvatar team={team} size={18} />,
+      })),
+    [teams]
   );
 
+  const [triggerIcon, triggerLabel] = useMemo(() => {
+    const firstSelectedSuggestion =
+      selectedTeams[0] && suggestedOptions.find(opt => opt.value === selectedTeams[0]);
+
+    const firstSelectedTeam =
+      selectedTeams[0] && teams.find(team => team.id === selectedTeams[0]);
+
+    if (firstSelectedSuggestion) {
+      return [<IconUser key={0} />, firstSelectedSuggestion.label];
+    }
+
+    if (firstSelectedTeam) {
+      return [
+        <TeamAvatar team={firstSelectedTeam} size={16} key={0} />,
+        `#${firstSelectedTeam.slug}`,
+      ];
+    }
+
+    return [
+      <IconUser key={0} />,
+      showMyTeamsDescription ? t('My Teams') : t('All Teams'),
+    ];
+  }, [selectedTeams, teams, showMyTeamsDescription]);
+
   return (
-    <Filter
-      fullWidth={fullWidth}
-      showMyTeamsDescription={showMyTeamsDescription}
-      header={
-        <InputWrapper>
-          <StyledInput
-            autoFocus
-            placeholder={t('Filter teams')}
-            onClick={event => {
-              event.stopPropagation();
-            }}
-            onChange={(event: React.ChangeEvent<HTMLInputElement>) => {
-              const search = event.target.value;
-              setTeamFilterSearch(search);
-              debouncedSearch(search);
-            }}
-            value={teamFilterSearch || ''}
-          />
-          {fetching && <StyledLoadingIndicator size={16} mini />}
-        </InputWrapper>
+    <CompactSelect
+      multiple
+      isClearable
+      isSearchable
+      isLoading={fetching}
+      menuTitle={t('Filter teams')}
+      options={
+        showSuggestedOptions
+          ? [
+              {value: '_suggested', label: t('Suggested'), options: suggestedOptions},
+              {value: '_teams', label: t('Teams'), options: teamOptions},
+            ]
+          : teamOptions
```

Comment:
Splitting this up into 2 sections, based on @robinrendle's design
![image](https://user-images.githubusercontent.com/44172267/169916195-178a2ff0-137d-4579-baf2-c41f92e759a3.png)

Reply:
cc @robinrendle